index.php

```diff
@@ -343,8 +362,16 @@ public function checkWritePermissions() {
 	 * @throws Exception when config.php can't be written
 	 */
 	public function setMaintenanceMode($state) {
+		$this->silentLog('[info] setMaintenanceMode("' . ($state ? 'true' : 'false') .  '")');
+
+		if($dir = getenv('NEXTCLOUD_CONFIG_DIR')) {
+			$configFileName = rtrim($dir, '/') . '/config.php';
+		} else {
+			$configFileName = __DIR__ . '/../config/config.php';
+		}
+		$this->silentLog('[info] configFileName ' . $configFileName);
+
 		/** @var array $CONFIG */
-		$configFileName = __DIR__ . '/../config/config.php';
 		require $configFileName;
```

Comment:
Add a check if the config file really exists, doesn't have to be the case with `NEXTCLOUD_CONFIG_DIR`.